spec/index.adoc

```diff
@@ -1654,6 +1654,51 @@ There is no restriction on who can invoke this method.
 
 This method takes no input and returns 32 pseudo-random bytes to the caller. The return value is unknown to any part of the IC at time of the submission of this call. A new return value is generated for each call to this method.
 
+[#ic-http_request]
+=== IC method `http_request`
+
+This method makes an HTTP request to a given URL and returns the HTTP response, possibly after a transformation.
+
+For security, the request is executed by multiple IC replicas (usually the entire subnet) and therefore must be _idempotent_. This means that the request must not change state at the remote server, or that the remote server has means to identify duplicated requests.
+
+The responses for all the requests must be, eventually, identical. However, it is possible that a web service would return slightly different responses for the same idempotent request. For example, it may include some unique identification or a timestamp that would vary across responses.
+
+For this reason, the calling canister can supply a transformation function, which is used by the IC to let the canister sanitize the responses from such unique values, and therefore eventually agree on an identical response. The transformation function is executed separately by each replica on the corresponding response it has received for the request. The final response will only be available to the calling canister if enough replicas have agreed on an identical response.
+
+Currently, only GET method is supported for HTTP requests.
+
+For security reasons, only HTTPS connections are allowed (URLs must start with `https://`). The IC uses industry-standard root CA lists to validate certificates of remote web servers.
+
+The cost of each call to the `http_request` method is `TBD` (the `transform` should also be charged).
+
+The maximal size of a request URL is 2048 bytes.
+
+The maximal size of a response is `TBD`. If a response is larger than this size, only the first `TBD` bytes will be returned. This size limit also applies to the value returned by the `transform` function.
```

Comment:
“also”? Or “only”? I'd expect that larger responses can be processed and trimmed down by the transform function, as we really only have a problem with large messages when they enter consensus, and that’s post-`transform`.

Reply:
@yotamhc replaced `TBD` with `2MiB`, we can modify this value later if needed, but for the first iteration, I think it's fine. Please see also Joachim's suggestion here. Is this something that we are planning to do now or maybe later, in the following iterations?

Reply:
@nomeata We also don't want to download too big responses (before the `transform`). We could have another limit for that, but opted to keep it simple for now. Do you think it's crucial to allow larger responses?